pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentSubscription.java

```diff
@@ -232,6 +276,29 @@ public void acknowledgeMessage(List<Position> positions, AckType ackType, Map<St
             if (log.isDebugEnabled()) {
                 log.debug("[{}][{}] Individual acks on {}", topicName, subName, positions);
             }
+            // Check if message is acknowledged by ongoing transaction.
+            if ((pendingAckMessages != null && pendingAckMessages.size() != 0) || pendingCumulativeAckMessage != null) {
+                synchronized (PersistentSubscription.this) {
+                    positions.forEach(position -> {
+                        checkArgument(position instanceof PositionImpl);
+                        // If single ack try to ack message in pending_ack status, fail the ack.
+                        if (pendingAckMessages != null && this.pendingAckMessages.contains(position)) {
+                            log.warn("[{}][{}] Invalid acks position conflict with an ongoing transaction:{}.",
+                                    topicName, subName, this.pendingCumulativeAckTxnId.toString());
+                            return;
```

Comment:
if consumer ack 1,2,3,4,5 and 1 is acknowledged by ongoing transaction, is 2,3,4,5 can not be acked?

Reply:
2, 3, 4, 5 will be acked, return in foreach will just skip to next item.